src/ssl/MemStats.cc

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 1996-2025 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+
+#include "base/PackableStream.h"
+#include "ssl/MemStats.h"
+
+#if USE_OPENSSL
+
+Ssl::MemStats::MemStats(const char *allocFunName, const char *freeFunName)
+    : allocFun(allocFunName), freeFun(freeFunName)
+{
+    allocSizes.logInit(20, 0, 1024*1024);
+}
+
+void
+Ssl::MemStats::alloc(const size_t bytes)
+{
+    numAllocs++;
+    if (bytes > maxAllocation)
+        maxAllocation = bytes;
+    allocSizes.count(bytes);
+}
+
+void
+Ssl::MemStats::dump(StoreEntry &e)
+{
+    PackableStream yaml(e);
+    assert(allocFun);
+    yaml << allocFun << "() calls: " <<  numAllocs << "\n";
```

Comment:
Do we need an indentation for these lines to be valid YAML for the "Current SSL memory usage" key?

Reply:
Yes, we need. Yaml does not allow tabs for it, so I added a 4-space indentation (though I saw Squid uses tabs in other reports).